src/node/rpc/node_frontend.h

```diff
@@ -441,12 +455,14 @@ namespace ccf
           rep.node_status = node_status;
           if (
             node_status == NodeStatus::TRUSTED ||
-            node_status == NodeStatus::LEARNER)
+            node_status == NodeStatus::LEARNER ||
+            node_status == NodeStatus::RETIRING)
```

Comment:
Do we really need this? AFAICS, this is for a new node that's waiting to be trusted by members but is retired in the meantime. So it might have been briefly trusted but never participated in the consensus (otherwise it would have already joined). So maybe joining nodes should never join if they're retiring?

Reply:
I added this because it happens during the reconfiguration tests. A node attempts to join, but is retired before it hears about it and thus can't decide whether it should submit the join RPC. So, I think it's necessary, but maybe the behaviour should be the same as for `NodeStatus::PENDING`?

Reply:
Move checks to function `is_taking_part_in_acking`. Add comment to explain why?